pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -617,12 +706,42 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         _require(normalizedSum == FixedPoint.ONE, Errors.NORMALIZED_WEIGHT_INVARIANT);
 
         _setMiscData(
-            _getMiscData().insertUint32(startTime, _START_TIME_OFFSET).insertUint32(endTime, _END_TIME_OFFSET)
+            _getMiscData().insertUint32(startTime, _WEIGHT_START_TIME_OFFSET).insertUint32(
+                endTime,
+                _WEIGHT_END_TIME_OFFSET
+            )
         );
 
         emit GradualWeightUpdateScheduled(startTime, endTime, startWeights, endWeights);
     }
 
+    function _startGradualSwapFeeChange(
+        uint256 startTime,
+        uint256 endTime,
+        uint256 startSwapFeePercentage,
+        uint256 endSwapFeePercentage
+    ) internal virtual {
+        _setSwapFeeData(startTime, endTime, startSwapFeePercentage, endSwapFeePercentage);
+
+        emit GradualSwapFeeUpdateScheduled(startTime, endTime, startSwapFeePercentage, endSwapFeePercentage);
+    }
+
+    function _setSwapFeeData(
+        uint256 startTime,
+        uint256 endTime,
+        uint256 startSwapFeePercentage,
+        uint256 endSwapFeePercentage
+    ) private {
+        _setMiscData(
+            _getMiscData()
+                .insertUint32(startTime, _FEE_START_TIME_OFFSET)
+                .insertUint32(endTime, _FEE_END_TIME_OFFSET)
+                .insertUint64(startSwapFeePercentage, _SWAP_FEE_PERCENTAGE_OFFSET)
```

Comment:
Related to https://github.com/balancer-labs/balancer-v2-monorepo/pull/1253/files#r857521915

This last line is a noop. `BasePool._setMiscData` explicitly clears out these top bits to stop us from changing the swap fee percentage so even if we fix the related issue then we'll get bad swap fee changes.

Reply:
I've added a failing testcase in [1f3a723](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1253/commits/1f3a72373b382a68ba2b51761e1a31c176931d43)